core/types/types.cc

```diff
@@ -614,10 +614,12 @@ TypePtr Types::resultTypeAsSeenFrom(Context ctx, TypePtr what, SymbolRef fromWha
 }
 
 TypePtr Types::getProcReturnType(Context ctx, const TypePtr &procType) {
```

Comment:
This seems to be a general utility function that has a general name. Having it do a `NilClass` substraction seems surprising given general name. Can we create a separate helper for this?

Reply:
Ya, I found it a little odd to do the subtraction here as well. Are you proposing something like the following (at the caller)?

```
auto tp = ...

blockReturnTp = getProcReturnType(subtractNilHelper(tp))
```

Reply:
yup, 
+  `subtractNilHelper` is `dropSubtypesOf(ctx, tp, Symbols::NilClass())`

Reply:
Made a `dropNil` helper function.